crates/astria-composer/src/searcher/geth_collector.rs

```diff
@@ -1,3 +1,14 @@
+//! `GethCollector` is responsible for fetching pending transactions submitted to rollup
+//! nodes and then passing them downstream for the searcher to process.
+//!
+//! It is responsible for fetching pending transactions submitted to the rollup Geth nodes and then
```

Comment:
```suggestion
//! [`GethCollector`] subscribes to pending transactions from a [go-ethereum](https://geth.ethereum.org) rollup node, and forwards them for more processing.
```
The rest of the information is orthogonal to this collector (the collector does not know about the searcher at this point, or if it can have multiple such collectors).